File: studyDecider.py
class StudyDecider:
    def __init__(self):
        self.num_of_classes = 0
        self.grades = []
        self.classes = []

    def update_elements(self, new_grades):
        for i in range(len(self.grades)):
            self.grades[i] += new_grades[i]

    def set_classes_num(self, num):
        return list(range(num))

    def rate_classes(self, arr, classes_num):
        # Bubble sort (descending order)
        for i in range(len(arr) - 1):
            for j in range(len(arr) - 1 - i):
                if arr[j] < arr[j + 1]:
                    arr[j], arr[j + 1] = arr[j + 1], arr[j]
                    classes_num[j], classes_num[j + 1] = classes_num[j + 1], classes_num[j]
        return arr

    def ask_questions(self, question, names, grades):
        print(question)
        for i, name in enumerate(names):
            am = int(input(f"{name}: "))
            grades[i] += am

    def link_lists(self, nums, lst):
        new_list = [None] * len(lst)
        for i, idx in enumerate(nums):
            new_list[idx] = lst[i]
        return new_list

    def print_list(self, grades):
        for g in grades:
            print(g)

    def print_slist(self, names):
        for n in names:
            print(n)

    def set_num_of_classes(self, num):
        self.num_of_classes = num


def main():
    sd = StudyDecider()

    num = int(input("Type how many subjects you have and want to study: "))
    if num == 1:
        print("Study this one subject..")
        return

    sd.set_num_of_classes(num)

    print("\nType names of subjects: ")
    names = [input().strip() for _ in range(num)]

    print("\n========================================================")
    print("Now you will rate some things carefully. You can always rerun if you change your mind.")
    print("========================================================\n")

    print(f"Rate from 1-{num} (with 1 being the lowest score):\n")
    grades = [0] * num

    questions = [
        f"How much stuff is there to study for each subject ({num}:a lot,1:basically nothing)",
        f"Which one is closer date-wise ({num}:really close,1:far far)",
        f"How well do you already know each class({num}:studied all year, 1:haven't ever studied it)",
        f"How easy is the class overall ({num}:very hard,1:very easy)",
        f"How hard are the exams ({num}:very hard,1:very easy)"
    ]

    for q in questions:
        sd.ask_questions(q, names, grades)

    classes_num = sd.set_classes_num(num)
    sd.rate_classes(grades, classes_num)
    final_list = sd.link_lists(classes_num, names)

    print("\n******************** computing.... ****************\n")
    print("Here’s what you graded in order from max points to least points:")
    sd.print_list(grades)

    print("\n========================================================")
    print("Here’s the order from MOST to LEAST significant class to study:")
    sd.print_slist(final_list)


if __name__ == "__main__":
    main()

File: StudyDecider.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Arrays;

public class StudyDecider{
	
	private int numOfClasses;
	private int[] grades = new int[numOfClasses];
	private String[] classes = new String[numOfClasses];
	
	
	public void updateElements(int[] newGrades){
		this.grades = grades;
		for(int i=0;i<grades.length;i++){
			grades[i] += newGrades[i];
		}
	}
	
	public void setClassesNum(int[] classesNum,int num){
		for(int i =0;i<num;i++){
			classesNum[i] = i;
		}
	}
	
	public int[] rateClasses(int[] arr, int[] classesNum){
		int[] sortedList = new int[arr.length];
		int count = 0;
		int temp;
		int temp2;
		for(int i =0;i<arr.length-1;i++){
			for(int j=0;j<arr.length-1-i;j++){
				if(arr[j]<arr[j+1]){
					temp = arr[j];
					temp2 = classesNum[j];
					
					arr[j] = arr[j+1];
					classesNum[j] = classesNum[j+1];
					
					arr[j+1] = temp;
					classesNum[j+1] = temp2;
				}				
			}
		}
		sortedList = arr;
		return sortedList;
	}
	
	public void askQuestions(String question,String[] names, int[] grades){
		System.out.println(question);
		Scanner sc = new Scanner(System.in);
		for(int i=0;i<names.length;i++){
			System.out.print(names[i]+": ");
			int am = sc.nextInt();
			grades[i] += am;
			System.out.println();
		}
	}
	
	public String[] linkLists(int[] nums, String[] list){
		String[] newList = new String[list.length];
		for(int i=0;i<nums.length;i++){
			int index = nums[i];
			newList[index] = list[i];
		}
		return newList;
	}
	
	public void printList(int[] grades){
		for(int i=0;i<grades.length;i++){
			System.out.println(grades[i]+" ");
		}
	}
	
	public void printSList(String[] grades){
		for(int i=0;i<grades.length;i++){
			System.out.println(grades[i]+" ");
		}
	}
	
	public void setNumOfClasses(int numOfClasses){
		this.numOfClasses = numOfClasses;
	}


	public static void main(String[] args){
		
		StudyDecider sd = new StudyDecider();
		Scanner sc = new Scanner(System.in);
		System.out.print("Type how many subjects you have and want to study: ");
		int num = sc.nextInt();
		sc.nextLine().trim();
		if(num==1){
			System.out.println("study this one subject..");
			System.exit(0);
		}
		sd.setNumOfClasses(num);
		
		System.out.println("\n\nType names of subjects: ");
		
		String[] names = new String[num];
		for(int i=0;i<num;i++){
			names[i] = sc.nextLine().trim();
		}
		
		System.out.println("========================================================");
		System.out.println("alright now we you will have to rate some things as carefully as u can but u can always rerun this if you change your mind");
		System.out.println("========================================================\n");
		
		System.out.println("Rate from 1-"+num+" (with 1 being the lowest score):\n\n");
		int[] grades = new int[num];
		for(int i=0;i<num;i++){
			grades[i]=0;
		}
		
		String[] questions = {"How much stuff is there to study for each subject ("+num+":a lot,1:basically nothing)","Which one is closer date-wise ("+num+":really close,1:far far)","How well do you already know each class("+num+": studied all year, 1:havent ever studied it)","How easy is the class overall ("+num+":very hard,1:very easy)","How hard are the exams ("+num+":very hard,1:very easy)"};
		
		for(int i=0;i<questions.length;i++){
			sd.askQuestions(questions[i],names,grades);
		}
	
		
		int[] classesNum = new int[num];		
		sd.setClassesNum(classesNum,num);
		sd.rateClasses(grades,classesNum);
		String[] finalList = sd.linkLists(classesNum,names);
		System.out.println("\n******************** computing.... ****************\n");
		System.out.println("heres what you graded in order from max points to least points:");
		sd.printList(grades);
		System.out.println("========================================================\n");
		System.out.println("Heres the order from MOST to LEAST significant class for u 2 study :)");
		sd.printSList(finalList);		
	
	}
}